src/groups/bmq/bmqu/bmqu_blob.cpp

```diff
@@ -494,12 +492,6 @@ int BlobUtil::readUpToNBytes(char*               buf,
         return -1;  // RETURN
     }
 
-    if (BSLS_PERFORMANCEHINT_PREDICT_UNLIKELY(
-            start.buffer() == blob.numDataBuffers() && length > 0)) {
-        BSLS_PERFORMANCEHINT_UNLIKELY_HINT;
-        return -2;  // RETURN
```

Comment:
`BlobPosition` is valid, we checked it already.
In this case, it is 100% valid to read 0 bytes in `readUpToNBytes` if we start reading from the rightmost blob position.
This allows to simplify and remove this extra check.